### PR TITLE
Dev remove lodash deps

### PR DESCRIPTION
**Pull request by privateOmega**
Changed the missionprogress array of objects into object of objects.
## Description
Refractored  missionprogress.js. Remove lodash dependencies
## Related Issue
https://github.com/DAVFoundation/missioncontrol/issues/133

## How Has This Been Tested?
node v8.9.4, npm v5.6.0
Ran npm test

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: server/controllers/StatusController.js [hunk 1]
```diff
@@ -41,7 +40,8 @@ const getStatus = async (req, res) => {
       const mission = latestMission;
       const vehicle = await getVehicle(latestMission.vehicle_id);
       const status = 'in_mission';
-      const currentStatus = _.find(missionProgress, {'status': vehicle.status});
+      const currentStatus = missionProgress[''+vehicle.status];
+      console.log(currentStatus);
```
Comment (TalAter) [inline server/controllers/StatusController.js]: This line should be deleted

File: server/controllers/StatusController.js [hunk 2]
```diff
@@ -41,7 +40,8 @@ const getStatus = async (req, res) => {
       const mission = latestMission;
       const vehicle = await getVehicle(latestMission.vehicle_id);
       const status = 'in_mission';
-      const currentStatus = _.find(missionProgress, {'status': vehicle.status});
+      const currentStatus = missionProgress[''+vehicle.status];
```
Comment (TalAter) [inline server/controllers/StatusController.js]: Why concatenate an empty string to it? There is no need to cast to string, it is done implicitly

Comment (privateOmega) [reply server/controllers/StatusController.js]: ohk. beginners mistake :+1:

File: server/controllers/StatusController.js (hunk 1 above)
Comment (privateOmega) [reply server/controllers/StatusController.js]: didn't notice that. will remove it

**Comment by privateOmega** [comment]
@TalAter I hope this resolves the issue.

File: server/simulation/missionProgress.js [hunk 3]
```diff
@@ -18,13 +18,7 @@ module.exports = [
       return elapsedSeconds > 2;
     },
   },
-  {
```
Comment (TalAter) [inline server/simulation/missionProgress.js]: The new changes from `master` brought in a new status that also needs to be handled.

**Comment by TalAter** [comment]
Looks great!

Thanks so much for the _**level up**_ of our code 👍

**Comment by privateOmega** [comment]
@TalAter Happy to help. Would be taking up another issue soon.

**Pull request merged**